src/data/cel.rs

```diff
@@ -0,0 +1,499 @@
+use crate::data::get_attribute;
+use crate::data::property::Path;
+use cel_interpreter::objects::{Map, ValueType};
+use cel_interpreter::{Context, Value};
+use cel_parser::{parse, Expression as CelExpression, Member, ParseError};
+use chrono::{DateTime, FixedOffset};
+use serde_json::Value as JsonValue;
+use std::collections::HashMap;
+use std::fmt::{Debug, Formatter};
+use std::sync::OnceLock;
+
+#[derive(Clone, Debug)]
+pub struct Expression {
+    attributes: Vec<Attribute>,
+    expression: CelExpression,
+}
+
+impl Expression {
+    pub fn new(expression: &str) -> Result<Self, ParseError> {
+        let expression = parse(expression)?;
+
+        let mut props = Vec::with_capacity(5);
+        properties(&expression, &mut props, &mut Vec::default());
+
+        let attributes = props
+            .into_iter()
+            .map(|tokens| {
+                let path = Path::new(tokens);
+                known_attribute_for(&path).unwrap_or(Attribute {
```

Comment:
When a expression contains `request.headers.X-MY-HEADER`, that path is not a known attribute. Hence, the attribute is read expecting a String and returned bytes are parsed as String. 

However, the prefix of it, i.e. `request.headers`, is well known, a `map<string, string>`, how could we leverage that to be smarter?

Reply:
Yes... tho probably for now I would read the entire map and let the user deal as such with it. 
Another way is to have `properties` follow maps, as we know we can ask envoy to resolve these for us. 
Cause technically, that's the [way](https://github.com/proxy-wasm/proxy-wasm-rust-sdk/blob/main/src/hostcalls.rs#L140-L164)

Reply:
Also, this `request.headers.X-MY-HEADER` is not a valid expression, it should be `request.headers["X-MY-HEADER"]`